### PR TITLE
Optimization for ConvNd if dropout=0.

### PR DESCRIPTION
**Pull request by gslama12**
As discussed in [#2153](https://github.com/huggingface/peft/pull/2153)

### CODE REVIEW

File: src/peft/tuners/lora/dora.py [hunk 1]
```diff
@@ -157,8 +157,9 @@ def forward(self, x, *, lora_A, lora_B, scaling, base_layer):
         # during backpropagation"
         weight_norm = weight_norm.detach()
         mag_norm_scale = magnitude / weight_norm
-        result_dora = (mag_norm_scale - 1) * (
-            self.conv_fn(
+
+        if base_result is None:
```
Comment (BenjaminBossan) [inline src/peft/tuners/lora/dora.py:161]: Wondering: Do we not need to account for the possibility that the base layer has a bias term, same as we do for linear?

https://github.com/huggingface/peft/blob/40fe16644697564c8d0b3996830d682630469e70/src/peft/tuners/lora/dora.py#L91-L94

**Review by BenjaminBossan — COMMENTED** [review]
Thanks for adding this DoRA optimization to conv layers. Generally, this looks good, but I have one comment about the bias term.

File: src/peft/tuners/lora/dora.py (hunk 1 above)
Comment (gslama12) [reply src/peft/tuners/lora/dora.py:161]: Yes i believe you are right. Since the previous case always used bias=None i think we should subtract it to keep the implementation consistent. I updated the code.

**Comment by BenjaminBossan** [comment]
Thanks for the update. Let's call `make style` or remove the trailing white space to make the linter happy.

**Comment by gslama12** [comment]
Should be fixed.

File: src/peft/tuners/lora/dora.py [hunk 2]
```diff
@@ -167,8 +168,12 @@ def forward(self, x, *, lora_A, lora_B, scaling, base_layer):
                 dilation=base_layer.dilation,
                 groups=base_layer.groups,
             )
-        ) + mag_norm_scale * lora_B(lora_A(x)) * scaling
+        else:
+            bias = base_layer.bias
+            if bias is not None:
+                base_result = base_result - bias
```
[the remark this answers is not in the document]
Comment (gslama12) [reply src/peft/tuners/lora/dora.py:174]: Looks good to me, thanks for the fix!

**Review by BenjaminBossan — APPROVED** [review]
Thanks for bringing this DoRA optimization to conv layers.
